x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/literal/interval/Interval.java

```diff
@@ -26,7 +26,8 @@
  * Unfortunately because the SQL interval type is not preserved accurately by the JDK TemporalAmount class
  * in both cases, the data type needs to be carried around as it cannot be inferred.
  */
-public abstract class Interval<I extends TemporalAmount> implements ConstantNamedWriteable, ToXContentObject, IntervalScripting {
+public abstract class Interval<I extends TemporalAmount> implements ConstantNamedWriteable, ToXContentObject, IntervalScripting,
+        Comparable<Interval<I>> {
```

Comment:
Why an interval has to be comparable?

Reply:
In particular, this is part of the change applied to the `SqlProtocolTestCase`, which performs an equality test between the value in the SELECT and same value (plus its corresponding type), but passed as a parameter; which in turns verifies that accepting parameters of the tested types works.

In general, this also makes comparisons between intervals possible. (Other DBs, like Postgres, also implement this, so it shouldn't be that far fetched.) Currently a simple SELECT with an intervals comparison returns `null` (b/c that's what the `Comparisons.compare` returns if types are neither `Comparable`, nor `Number`; but this might be a different topic).

As far as the Java backing types go, it's a split case:
- `Duration` also implements `Comparable`;
- `Period` thought, doesn't, but that's because daylight savings are taking into account with the day component; however, this only backs up YEAR-MONTH intervals - i.e. no day component - so the extension in the patch is safe for these intervals.

Reply:
I would fix it in a separate PR to address the comparison between intervals and add proper testing.
Even more so considering `Period` doesn't implement `Comparable`.
In testing we only use the equality not comparison check . 

If comparing Period proves to be tricky, I propose handling this by preventing the comparison to be done.